Runtime/Setup/e2esetup.js

```diff
@@ -255,107 +257,124 @@ function setup_push(callback) {
+      setup_auth('twitter', clientId, clientSecret, done);
+    },
+
+    function (done) {
+      var clientId = nconf.get('aadClientId');
+      var tenant = nconf.get('aadTenant');
+      setup_aad(clientId, tenant, done);
+    }
   ], callback);
 }
 
-function setup_auth_AllProviders(callback) {  
-	async.series([
-		function(done) {
-			var clientId = nconf.get('googleClientId');  
-			var clientSecret = nconf.get('googleClientSecret');  
-			setup_auth('google',clientId,clientSecret,done);
-		},
-	
-		function(done) {
-			var clientId = nconf.get('msClientId');  
-			var clientSecret = nconf.get('msClientSecret');  			
-			setup_auth('microsoftaccount',clientId,clientSecret,done);
-		},
-	
-		function(done) {
-			var clientId = nconf.get('fbClientId');  
-			var clientSecret = nconf.get('fbClientSecret');
-			setup_auth('facebook',clientId,clientSecret,done);
-		},
-	
-		function(done) {
-			var clientId = nconf.get('twitterApiKey');  
-			var clientSecret = nconf.get('twitterApiSecret');   
-			setup_auth('twitter',clientId,clientSecret,done);
-		}
-	], callback);  
+function setup_auth(authprovider, clientId, clientSecret, callback) {
+  if (!clientId || !clientSecret) {
+    return callback();
+  }
+  var indent = '   ';
+  process.stdout.write(indent + 'Setting up ' + authprovider + ' auth. clientId (' + clientId + ') clientsecret (' + clientSecret + ')...');
+  if (authprovider == 'microsoftaccount') {
+    var msPackageSID = nconf.get('msPackageSID');
+    invoke_scripty(indent, 'mobile auth ' + authprovider + ' set --packageSid ' + msPackageSID + ' ' + nconf.get('name') + ' ' + clientId + ' ' + clientSecret, callback);
+  } else {
+    invoke_scripty(indent, 'mobile auth ' + authprovider + ' set ' + nconf.get('name') + ' ' + clientId + ' ' + clientSecret, callback);
+  }
 }
 
-function setup_auth(authprovider,clientId,clientSecret,callback) {  
-  if (!clientId) {
+function setup_aad(clientId, tenant, callback) {
+  if (!clientId || !tenant) {
     return callback();
-  }  
-  if (!clientSecret) {
+  }
+  if (!validator.isUUID(clientId)) {
+    console.log('   Warn: supplied ClientId is not a GUID, skipping AAD Auth...'.yellow.bold);
     return callback();
   }
-  process.stdout.write('   Setting up '+ authprovider + ' auth. clientId (' + clientId + ') clientsecret (' + clientSecret + ')...');
-  if(authprovider == 'microsoftaccount') {
-	var msPackageSID = nconf.get('msPackageSID');  		
-	scripty.invoke('mobile auth '+ authprovider +' set --packageSid ' + msPackageSID + ' ' + nconf.get('name') + ' ' + clientId + ' '+ clientSecret, function(err, results) {
-		if (!err) {
-		console.log(' OK'.green.bold);
-		}
-		callback(err);
-	});
-  }  
-  else {
-	scripty.invoke('mobile auth '+ authprovider +' set ' + nconf.get('name') + ' ' + clientId + ' '+ clientSecret, function(err, results) {
-		if (!err) {
-		console.log(' OK'.green.bold);
-		}
-		callback(err);
-	});
+  if (!validator.isFQDN(tenant)) {
```

Comment:
if the tenant is not valid IIS throws 500s. To avoid these errors adding minor validation to ensure that the tenant is at least FQDN